src/posts/posts.controller.ts

```diff
@@ -0,0 +1,185 @@
+import {
+    Body,
+    Controller,
+    DefaultValuePipe,
+    Delete,
+    Get,
+    Param,
+    ParseIntPipe,
+    Patch,
+    Post,
+    Query,
+    Res,
+    StreamableFile,
+    UploadedFile,
+    UseGuards,
+} from '@nestjs/common';
+import { PostsService } from './posts.service';
+import { CreatePostDto } from './dto/create-post.dto';
+import { UpdatePostDto } from './dto/update-post.dto';
+import { JwtAuthGuard } from '../../common/guards/jwt-auth.guard';
+import { UsersRoleGuard } from '../../common/guards/users-role.guard';
+import { FilesInterceptor } from '../../common/interceptors/file-interceptor';
+import { fileValidationPipe } from '../../common/decorators/upload-files';
+import { CurrentUser } from '../../common/decorators/current-user';
+import { UserEntity } from '../users/user.entity';
+import { UserRoles } from '../../common/decorators/user.roles';
+import { Role } from '../../common/enums/users.role';
+import { Response } from 'express';
+import {
+    ApiBody,
+    ApiConsumes,
+    ApiCreatedResponse,
+    ApiOkResponse,
+    ApiOperation,
+    ApiParam,
+    ApiQuery,
+    ApiTags,
+} from '@nestjs/swagger';
+import { PostEntity } from './post.entity';
+import { ApiFileResponse } from '../../common/swagger/swagger.decorators';
+import { FilesService } from '../files/files.service';
+import { Pagination } from 'nestjs-typeorm-paginate';
+import { Order } from '../../common/enums/pagination.order';
+import { OrderCondition } from '../../common/enums/order.condition';
+
+@ApiTags('Posts')
+@UseGuards(JwtAuthGuard, UsersRoleGuard)
+@Controller('posts')
+export class PostsController {
+    constructor(private readonly postsService: PostsService, private filesService: FilesService) {}
+
+    @ApiOperation({ summary: 'Create Post' })
+    @ApiCreatedResponse({
+        type: PostEntity,
+        description: 'Created Post Response',
+    })
+    @ApiConsumes('multipart/form-data')
+    @FilesInterceptor('file')
+    @ApiBody({
+        type: CreatePostDto,
+        description: 'Create Post Fields',
+    })
+    @Post()
+    async createPost(
+        @Body() body: CreatePostDto,
+        @UploadedFile(fileValidationPipe('image/jpeg')) file: Express.Multer.File,
+        @CurrentUser() currentUser: UserEntity,
+    ): Promise<PostEntity> {
+        return this.postsService.createPost(body, currentUser, file);
+    }
+
+    @ApiOperation({ summary: 'Get All Posts' })
+    @ApiOkResponse({
+        type: PostEntity,
+        description: 'Posts Response',
+    })
+    @ApiQuery({ name: 'page', type: 'number', required: false })
+    @ApiQuery({ name: 'limit', type: 'number', required: false })
+    @ApiQuery({ name: 'order', enum: Order, required: false, example: 'ASC' })
+    @ApiQuery({ name: 'condition', enum: OrderCondition, required: false, example: 'ID' })
+    @Get()
+    async findAll(
+        @Query('page', new DefaultValuePipe(1), ParseIntPipe) page: number = 1,
```

Comment:
Please check if it is possible to avoid using hardcoded values 1 and 10, probably they can be defined in the constant file.